### PR TITLE
feat (DPLAN-3796) Adjust filename

### PR DESCRIPTION
**Pull request by sierrazamudiodemos**
### Ticket
DPLAN-3796

Description:
1. The exported word doc per stn can be adjusted based on templates.
2. The method getFileName was introduced to adjust the name of the doc. Accepted placeholders are: '{ID} {NAME}{EINGANSNR}'
3. When no placeholder is given, the default one '{ID}-{NAME}-{EINGANSNR}' is taken

### How to review/test
1. FE is not integrated yet, but you can adjust the controller to play around with the template (pretending as if the FE would send it)
2. The controller to update ist: demosplan/DemosPlanCoreBundle/Controller/Segment/SegmentsExportController.php after line 65 and after line 75 you can add $fileNameTemplate = '{ID}-{NAME}-{EINGANSNR}' or set any combination of these placeholders
3. Test that when files have the same name, then the ID of the stn is added at the end (applicable for doc files that are in the zip folder)
4. Test that when the placeholder is empty, the default one is used

Delete the checkbox if it doesn't apply/isn't necessary.

- [x] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php [hunk 1]
```diff
@@ -55,6 +56,58 @@ public function __construct(
...
+        $defaultTemplateName = '{ID}-{NAME}-{EINGANSNR}';
+        $templateName = $templateName ?: $defaultTemplateName;
+
+        $externalId = $this->getExternalId($statement);
+        $authorSourceName = $this->getAuthorName($statement);
+        $internId = $this->getInternalId($statement);
+
+        // Replace placeholders with actual values from the $statement object
+        $fileName = str_replace(
+            ['{ID}', '{NAME}', '{EINGANSNR}'],
```
Comment (spiess-demos) [inline demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php:70]: Minor: Typo (should be EINGAN**G**SNR) - applies to several occurrences.

File: translations/messages+intl-icu.de.yml [hunk 2]
```diff
@@ -3553,6 +3553,8 @@ submitter.name: "Name des/der Einreicher*in"
 submitter.name.gendered: "Name des/der Einreicher*in"
 submitter.type: "Art des/der Einreicher*in"
 statement.name_source.unknown: "Einreichende Person unbekannt"
+statement.intern_id.unknown: "Eingangsnummer unbekannt"
+statement.extern_id.unknown: "ID unbekannt"
```
Comment (spiess-demos) [inline translations/messages+intl-icu.de.yml:3557]: nitpick: alphabetic sorting of keys.

File: demosplan/DemosPlanCoreBundle/Controller/Segment/SegmentsExportController.php [hunk 3]
```diff
@@ -74,11 +73,7 @@ static function () use ($procedure, $statement, $exporter, $tableHeaders) {
             }
         );
 
-        $filename = $slugify->slugify($procedure->getName())
-            .'-'
-            .$statement->getExternId().'.docx';
-
-        $this->setResponseHeaders($response, $filename);
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Controller/Segment/SegmentsExportController.php:81]: '.docx' is missing. I don't get a docx file when I export detail view.

**Review by breierDemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/DataGenerator/Factory/Statement/StatementMetaFactory.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * @phpstan-method static list<StatementMeta&Proxy<StatementMeta>> createSequence(iterable|callable $sequence)
+ * @phpstan-method static list<StatementMeta&Proxy<StatementMeta>> findBy(array $attributes)
+ * @phpstan-method static list<StatementMeta&Proxy<StatementMeta>> randomRange(int $min, int $max, array $attributes = [])
+ * @phpstan-method static list<StatementMeta&Proxy<StatementMeta>> randomSet(int $number, array $attributes = [])
+ */
+final class StatementMetaFactory extends PersistentProxyObjectFactory
+{
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#factories-as-services
+     */
```
[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/DataGenerator/Factory/Statement/StatementMetaFactory.php]: done 7f8dd8ff366febffbdee4ae16e419415e90f8bdc 7b38919683463b47bca1fce3eec9e567228d4890

File: demosplan/DemosPlanCoreBundle/DataGenerator/Factory/Statement/StatementMetaFactory.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        ];
+    }
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#initialization
+     */
+    protected function initialize(): static
+    {
+        return $this
+            // ->afterInstantiate(function(StatementMeta $statementMeta): void {})
```
[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/DataGenerator/Factory/Statement/StatementMetaFactory.php]: done 7f8dd8ff366febffbdee4ae16e419415e90f8bdc 7b38919683463b47bca1fce3eec9e567228d4890

File: demosplan/DemosPlanCoreBundle/Controller/Segment/SegmentsExportController.php (hunk 3 above)
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/Controller/Segment/SegmentsExportController.php:81]: done 2d4964e27d8185448811b46108ec07f8f538a725

File: demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php [hunk 6]
```diff
@@ -55,6 +56,58 @@ public function __construct(
         $this->slugify = $slugify;
     }
```
[the remark this answers is not in the document]
Comment (breierDemos) [reply demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php:58]: Maybe something like: 
- (SegmentExport)FileNameGenerator
- (SegmentExport)FileNameFormatter
- (SegmentExport)FileNameBuilder
- (SegmentExport)FileNameUtil
- (SegmentExport)FileNameService

[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php]: Done, I think the easiest is to refresh the PR to see the file changes:
- I created a new file SegmentExporterFileNameGenerator with the 4 methods
- I created a new file for tests SegmentExporterFileNameGeneratorTest
- I removed it from SegmentsByStatementsExporter and moved related tests as well

**Review by breierDemos — APPROVED** [review]
Looks Good! Thanks!

File: demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (breierDemos) [reply demosplan/DemosPlanCoreBundle/Logic/Segment/SegmentsExporter.php]: nice

**Pull request merged**